src/components/authorization/signUp/SignUp.jsx

```diff
@@ -0,0 +1,144 @@
+import "../authorization.css";
+import {
+  BiDoorOpenFill,
+  BiEyeFill,
+  BiEyeSlashFill,
+} from "assets/icons/Icons";
+import { Link } from "react-router-dom";
+import { useState, useEffect } from "react";
+
+
+function SignUp() {
+  useEffect(() => {
+    const timeOut = setTimeout(() => {
+      setError({ msg: "", state: false });
+    }, 5000);
+    return () => clearTimeout(timeOut);
+  });
+
+  const [signUpDetails, setSignUpDetails] = useState({
+    fullName: "",
+    email: "",
+    password: "",
+    confirmPassword: "",
+  });
+
+  const validation = /^(?=.*\d)(?=.*[a-z])([!@#$%]*).{5,}$/;
+  const [error, setError] = useState({ msg: "", state: false });
+  const [showPassword, setShowPassword] = useState({
+    password: true,
+    confirmPassword: true,
+  });
+
+  const signUpValidator = (event, signUpDetails) => {
+    event.preventDefault();
+    const { fullName, email, password, confirmPassword } = signUpDetails;
```

Comment:
since signUpDetails is your state variable you can access it directly and it need not be passed as an argument for signUpValidator function

Reply:
Yeah, will do. Thanks for pointing out.